### PR TITLE
Make hidden input _charset_ check case insensitive

### PR DESCRIPTION
**Pull request by josepharhar**
The spec for this is changing:
https://github.com/whatwg/html/issues/5914

### CODE REVIEW

File: html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html [hunk 1]
```diff
@@ -19,7 +20,8 @@
     const params = new URL(event.data).searchParams;
 
     assert_true(params.get("_charset_") != "", "lowercase valid");
```
Comment (annevk) [inline html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html:22]: Can you change this and the others to assert_equals with an actual expected value?

**Comment by domenic** [comment]
Ping to get this finished up and merged

File: html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html (hunk 1 above)
Comment (josepharhar) [reply html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html]: done

**Comment by josepharhar** [comment]
> Ping to get this finished up and merged

My bad, I addressed the comment

File: html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html [hunk 2]
```diff
@@ -19,10 +19,10 @@
...
     const params = new URL(event.data).searchParams;
 
-    assert_true(params.get("_charset_") != "", "lowercase valid");
-    assert_true(params.get("_CHARSET_") != "", "uppercase valid");
-    assert_true(params.get("_ChArSeT_") != "", "mixed case invalid");
-    assert_true(params.get("_charſet_") == "", "non-ASCII invalid");
+    assert_not_equals(params.get("_charset_"), "", "lowercase valid");
+    assert_not_equals(params.get("_CHARSET_"), "", "uppercase valid");
+    assert_not_equals(params.get("_ChArSeT_"), "", "mixed case invalid");
+    assert_equals(params.get("_charſet_"), "", "non-ASCII invalid");
```
Comment (domenic) [inline html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html:25]: I think the idea was to change them from not_equals to actually testing `assert_equals` against the correct value. Which I think is "UTF-8" in this case? Or maybe "utf-8"?

Comment (josepharhar) [reply html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html:25]: ahh that makes more sense, done!

**Review by domenic — APPROVED** [review]

File: html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html [hunk 3]
```diff
@@ -18,9 +19,10 @@
   addEventListener("message", this.step_func_done(event => {
     const params = new URL(event.data).searchParams;
 
-    assert_true(params.get("_charset_") != "", "lowercase valid");
-    assert_true(params.get("_ChArSeT_") == "", "mixed case invalid");
-    assert_true(params.get("_charſet_") == "", "non-ASCII invalid");
+    assert_equals(params.get("_charset_"), "UTF-8", "lowercase valid");
+    assert_equals(params.get("_CHARSET_"), "UTF-8", "uppercase valid");
+    assert_equals(params.get("_ChArSeT_"), "UTF-8", "mixed case invalid");
```
Comment (ghostd) [inline html/semantics/forms/the-input-element/hidden-charset-case-sensitive.html:24]: The comment should be "mixed case valid", isn't it?

**Pull request merged**